### PR TITLE
Implement ABI validate_address()

### PR DESCRIPTION
**Pull request by litchipi**
Closes: https://github.com/massalabs/massa/issues/3644
Closes: https://github.com/massalabs/massa-sc-runtime/issues/142
Massa-sc-runtime PR: https://github.com/massalabs/massa-sc-runtime/pull/238

* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-execution-worker/src/interface_impl.rs
Comment (sydhds) [inline massa-execution-worker/src/interface_impl.rs:537]: Any reason to return here a Result as a bool looks enough?

**Review by sydhds — CHANGES_REQUESTED** [review]
Looks good to me but there is no unit test. Is this intentional?

**Comment by litchipi** [comment]
Not intentional, I'll try to create some

File: massa-execution-worker/src/interface_impl.rs
Comment (litchipi) [reply massa-execution-worker/src/interface_impl.rs:537]: Indeed, was just mimicking the other functions, but can take it out

Comment (litchipi) [reply massa-execution-worker/src/interface_impl.rs:537]: No I can't, because in the trait definition in `src/types.rs`, it fails as apparently `unimplemented!` will type check as `Result` for clippy

**Comment by litchipi** [comment]
(Waiting for https://github.com/massalabs/massa-as-sdk/pull/164 to be merged before creating a unit test)

**Comment by litchipi** [comment]
@sydhds Added a unit test, but I copied it from `datastore_manipulations` a bit stupidly so please could you review the code and tell me what are the un-necessary parts of it ?

Also I'll wait for your review on https://github.com/massalabs/massa-unit-tests-src/pull/33 before merging it (and this one also)

File: massa-execution-worker/src/interface_impl.rs
Comment (sydhds) [inline massa-execution-worker/src/interface_impl.rs:538]: Wondering if it should not be is_ok() instead of is_err(). validate should return true if all went well no?

Comment (litchipi) [reply massa-execution-worker/src/interface_impl.rs]: Oops, I changed it but forgot to push it

File: massa-execution-worker/src/tests/scenarios_mandatories.rs
Comment (sydhds) [inline massa-execution-worker/src/tests/scenarios_mandatories.rs:1952]: Why not matching directly the wanted length?

File: massa-execution-worker/src/tests/scenarios_mandatories.rs
Comment (sydhds) [inline massa-execution-worker/src/tests/scenarios_mandatories.rs:1953]: Still needed?

Comment (litchipi) [reply massa-execution-worker/src/tests/scenarios_mandatories.rs:1953]: Well as it's only printed if the test fails, I feel it's nice to see why it failed, but no it can be deleted if you think it's better

File: massa-execution-worker/src/tests/scenarios_mandatories.rs
Comment (litchipi) [reply massa-execution-worker/src/tests/scenarios_mandatories.rs]: Fixed

File: massa-execution-worker/src/tests/scenarios_mandatories.rs
Comment (sydhds) [reply massa-execution-worker/src/tests/scenarios_mandatories.rs:1953]: If this is useful when the test fail maybe it should be include in the assert msg?

**Comment by litchipi** [comment]
As I merged the unit tests for #3640 in the same unit tests release, it'll have to wait till it's merged before this can pass

**Review by AurelienFT — APPROVED** [review]

**Comment by AurelienFT** [comment]
@sydhds can you resolve your comment ? as this PR blocks for testnet_21 to work well

**Review by sydhds — APPROVED** [review]

**Pull request merged**
